test/vaadin-combo-box-dropdown.html

```diff
@@ -112,6 +123,44 @@
         });
       });
     });
+
+    describe('in modal dialog', () => {
+      let backdrop;
+      let dialog;
+      let comboBox;
+
+      beforeEach((done) => {
+        dialog = fixture('in-modal-dialog');
+        dialog.opened = true;
+        requestAnimationFrame(() => {
+          comboBox = dialog.$.overlay.querySelector('vaadin-combo-box');
+          backdrop = dialog.$.overlay.$.backdrop;
+          done();
+        });
+      });
+
+      it('should not close the dialog on outside click', (done) => {
+        comboBox.open();
+
+        requestAnimationFrame(() => {
+          // For this test we need to remove the focus from the input element without closing the combobox dropdown,
+          // otherwise the issue is not reproducible, because the combo box overlay will be closed by the focus out
+          // event, rather than the click event. While this shouldn't never be the case for the combo box itself,
+          // this is a valid case for other components using combo box, such as time picker.
+          comboBox.addEventListener("focusout", e => {
+            e.stopPropagation()
+          }, true);
+          comboBox.blur();
+          expect(comboBox.opened).to.be.true;
```

Comment:
For some reason that required more creativity here to make sure that the combo box does not have focus. Maybe the test in the 21 branch does not work as intended. But shouldn't be a big deal, if it's obsolete in like a month or so.